File: Models/VehicleService.cs
public class VehicleService
{
    public int Id { get; set; }

    public int VehicleId { get; set; }
    public Vehicle? Vehicle { get; set; }

    public int? ServiceId { get; set; }
    public Service? Service { get; set; }

    public int? MechanicId { get; set; }
    public User? Mechanic { get; set; }
    public string? CustomerDescription { get; set; }
    public string? MechanicNote { get; set; }

    public DateTime RequestedAt { get; set; }
    public DateTime? CompletedAt { get; set; }

    public ServiceStatus Status { get; set; } = ServiceStatus.Requested;

    public decimal FinalPrice { get; set; }
}

public enum ServiceStatus
{
    Requested,
    InService,
    Completed,
    Cancelled
}

File: frontend/src/types/vehicleService.ts
export type ServiceStatus = 'Requested' | 'InService' | 'Completed' | 'Cancelled'

export type VehicleService = {
  id: number
  vehicleId: number
  vehicleMake: string | null
  vehicleModel: string | null
  serviceId: number | null
  mechanicId: number | null
  customerDescription: string | null
  mechanicNote: string | null
  requestedAt: string
  completedAt: string | null
  status: ServiceStatus
  finalPrice: number | null
}

export type UpdateVehicleServiceWorkRequest = {
  status?: ServiceStatus
  mechanicNote?: string
  finalPrice?: number
}

export type UpdateVehicleServiceRequest = {
  serviceId?: number
  mechanicId?: number
  customerDescription?: string | null
  mechanicNote?: string | null
  status?: ServiceStatus
  finalPrice?: number
}

export type VehicleServiceQuery = {
  vehicleId?: number
  mechanicId?: number
  serviceStatuses?: ServiceStatus[]
}

export type CreateVehicleServiceRequest = {
  vehicleId: number
  serviceId?: number | null
  customerDescription?: string | null
}
